### PR TITLE
[DOC, MRG] Minor documentation improvements and remove glossary entry for array-like

### PR DESCRIPTION
**Pull request by mscheltienne**
A couple of minor formatting fixes, and I removed the entry for `array-like` in favor of the one defined in the `numpy` glossary.

Also looking if I can figure out which CSS variable to override to format `terms` in the same way as `py:obj`. 

Screenshot 2022-09-26 at 13 59 10

### CODE REVIEW

File: mne/utils/docs.py [hunk 1]
```diff
@@ -1963,18 +1963,18 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
...
 """
 docdict['method_plot_psd_auto'] = _method_psd.format(
-    " | 'auto'",
+    " | ``'auto'``",
     (" ``'auto'`` (default) uses Welch's method for continuous data and "
      "multitaper for :class:`~mne.Epochs` or :class:`~mne.Evoked` data.")
 )
 docdict['method_psd'] = _method_psd.format('', '')
-docdict['method_psd_auto'] = _method_psd.format(" | 'auto'", '')
+docdict['method_psd_auto'] = _method_psd.format(" | ``'auto'``", '')
```
Comment (agramfort) [inline mne/utils/docs.py:1977]: are these changes consistent with the rest of the docstrings?

**Review by agramfort — COMMENTED** [review]
@mscheltienne can you share a link to the rendered doc for us to see how it impacts doc? 🙏

File: mne/utils/docs.py (hunk 1 above)
Comment (mscheltienne) [reply mne/utils/docs.py:1977]: They are, if I did not make a mistake. I'll check and share the links to the render doc as soon as it's built.

**Comment by mscheltienne** [comment]
Spectrum class: removes the whitespace before the `[k]` reference, add the missing `footbibliography` directives, and format the argument `method` with double backticks.
- before: https://mne.tools/dev/generated/mne.time_frequency.Spectrum.html?highlight=spectrum#footcite-welch1967
- after: https://output.circle-artifacts.com/output/job/c6f49603-b97d-4aef-bece-78550cf5b6d5/artifacts/0/dev/generated/mne.time_frequency.Spectrum.html#mne.time_frequency.Spectrum

Same for EpochsSpectrum class.
- before: https://mne.tools/dev/generated/mne.time_frequency.EpochsSpectrum.html?highlight=epochspectrum
- after: https://output.circle-artifacts.com/output/job/c6f49603-b97d-4aef-bece-78550cf5b6d5/artifacts/0/dev/generated/mne.time_frequency.EpochsSpectrum.html#mne.time_frequency.EpochsSpectrum

Add references and footbiliography to `psd_array_welch` and `psd_array_multitaper` functions:
- welch: https://output.circle-artifacts.com/output/job/c6f49603-b97d-4aef-bece-78550cf5b6d5/artifacts/0/dev/generated/mne.time_frequency.psd_array_welch.html#mne.time_frequency.psd_array_welch
- multitaper: https://output.circle-artifacts.com/output/job/c6f49603-b97d-4aef-bece-78550cf5b6d5/artifacts/0/dev/generated/mne.time_frequency.psd_array_multitaper.html#mne.time_frequency.psd_array_multitaper

Good to merge on my end, I haven't figured out which CSS variable had to be changed to format the `terms` in the same way as `py:obj`. I'll look into it again in another PR.

**Review by agramfort — APPROVED** [review]
@drammock ok for you?

File: mne/utils/docs.py (hunk 1 above)
Comment (drammock) [reply mne/utils/docs.py:1977]: in fact we're not systematic about using double-backticks in docstrings when specifying specific strings allowed as parameter values.  In a few cases they are mandatory (I think if it's a tuple maybe?).  But in most cases we don't include the double backticks (see, for example, `outlines : 'head' | 'skirt' | dict | None` or `ch_type : ‘mag’ | ‘grad’ | ‘planar1’ | ‘planar2’ | ‘eeg’ | None`

However, it does look better the way you've done it here, so maybe we should slowly move toward including double-backticks.

**Pull request merged**